teuthology/orchestra/remote.py

```diff
@@ -247,9 +245,9 @@ def sh(self, script, **kwargs):
             kwargs['stdout'] = BytesIO()
         if 'args' not in kwargs:
             kwargs['args'] = script
-        proc=self.run(**kwargs)
-        out=proc.stdout.getvalue()
-        return ensure_str(out)
+        proc = self.run(**kwargs)
+        out = proc.stdout.getvalue()
+        return out.decode()
```

Comment:
There should be a check if the stdout is BytesIO or StringIO to know if we need to apply or no `decode`.

Reply:
fixed.